### PR TITLE
ASoC: SOF: Intel: hda: report error only for the last ROM init iteration

### PR DESCRIPTION
**Pull request by ranj063**
The FW boot sequence includes multiple attempts for ROM init.
When it does take more than one attempt, we should not log the
errors encountered during the failed attempts and only log them
during the final iteration.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>

### CODE REVIEW

**Comment by ranj063** [comment]
SOFCI TEST

File: sound/soc/sof/intel/hda-loader.c [hunk 1]
```diff
@@ -329,25 +337,25 @@ int hda_dsp_cl_boot_firmware(struct snd_sof_dev *sdev)
 
 	/* try ROM init a few times before giving up */
 	for (i = 0; i < HDA_FW_BOOT_ATTEMPTS; i++) {
+		dev_dbg(sdev->dev,
+			"Attempting iteration %d of Core En/ROM load...\n", i);
```
Comment (lyakh) [inline sound/soc/sof/intel/hda-loader.c:341]: nit-pick: I think the original way of reporting *here* was a bit better: only `dev_dbg()` if an iteration fails, if the first iteration already succeeds, no need to print anything?

File: sound/soc/sof/intel/hda-loader.c [hunk 2]
```diff
@@ -90,7 +90,8 @@ static int cl_dsp_init(struct snd_sof_dev *sdev, const void *fwdata,
 	/* step 1: power up corex */
 	ret = hda_dsp_core_power_up(sdev, chip->cores_mask);
 	if (ret < 0) {
-		dev_err(sdev->dev, "error: dsp core 0/1 power up failed\n");
+		if (iteration == HDA_FW_BOOT_ATTEMPTS)
+			dev_err(sdev->dev, "error: dsp core 0/1 power up failed\n");
```
Comment (kv2019i) [inline sound/soc/sof/intel/hda-loader.c:94]: @ranj063  Hmm, how about making these all dev_warn() (without "error" in message) in cl_dsp_init. It's still suspicious that a boot attempt fails, so I think a warning is appropriate?

File: sound/soc/sof/intel/hda-loader.c [hunk 3]
```diff
@@ -329,25 +337,25 @@ int hda_dsp_cl_boot_firmware(struct snd_sof_dev *sdev)
...
+				  stripped_firmware.size, tag, i + 1);
 
 		/* don't retry anymore if successful */
 		if (!ret)
 			break;
+	}
 
-		dev_dbg(sdev->dev, "iteration %d of Core En/ROM load failed: %d\n",
+	if (i == HDA_FW_BOOT_ATTEMPTS) {
+		dev_err(sdev->dev, "error: dsp init failed after %d attempts with err: %d\n",
```
Comment (kv2019i) [inline sound/soc/sof/intel/hda-loader.c:352]: So warnings in cl_dsp_init(), and dev_err here if all attempts failed.

File: sound/soc/sof/intel/hda-loader.c (hunk 1 above)
Comment (ranj063) [reply sound/soc/sof/intel/hda-loader.c:341]: @lyakh @kv2019i I considered both your suggestions but unfortunately our CI scripts looks for words like error, failed, panic etc and even if I remove the word error, I'd still have to reword it to remove the word failed.

**Comment by plbossart** [comment]
@ranj063 do you happen to know if the failure on UP2_NOCODEC is related to this? Wondering if this is a CI issue.

**Comment by ranj063** [comment]
> @ranj063 do you happen to know if the failure on UP2_NOCODEC is related to this? Wondering if this is a CI issue.

@plbossart I am scratching my head about this failure too. My changes would only affect the bootflow. Let me dig into this

**Comment by plbossart** [comment]
> > @ranj063 do you happen to know if the failure on UP2_NOCODEC is related to this? Wondering if this is a CI issue.
> 
> @plbossart I am scratching my head about this failure too. My changes would only affect the bootflow. Let me dig into this

To be honest the recent CI results are 'too good to be true'. Not a single failure, and then this? Humm...

**Comment by kv2019i** [comment]
SOFCI TEST

**Comment by kv2019i** [comment]
Hmm, I was about to merge already, but then I checked past runs and the up2-nocodec has failed twice with this PR, while I did not see failures for others PRs (for this test). But based on code, I can't see how this could impact. Let's do one more round.

**Comment by ranj063** [comment]
> Hmm, I was about to merge already, but then I checked past runs and the up2-nocodec has failed twice with this PR, while I did not see failures for others PRs (for this test). But based on code, I can't see how this could impact. Let's do one more round.

@kv2019i I ran it the second time to see if I could reproduce it again and it did. But the error has disappeared now. Im not sure what to think of it

**Comment by kv2019i** [comment]
@ranj063 It seems the problem has existed before and is tracked here:
https://github.com/thesofproject/sof/issues/2921

... so proceeding with merge.
